src/main/kotlin/cacophony/controlflow/functions/CallableHandlerImpl.kt

```diff
@@ -0,0 +1,162 @@
+package cacophony.controlflow.functions
+
+import cacophony.controlflow.*
+import cacophony.controlflow.CFGNode.MemoryAccess
+import cacophony.controlflow.CFGNode.RegisterUse
+import cacophony.controlflow.generation.Layout
+import cacophony.controlflow.generation.generateLayoutOfVirtualRegisters
+import cacophony.semantic.analysis.AnalyzedFunction
+import cacophony.semantic.analysis.EscapeAnalysisResult
+import cacophony.semantic.analysis.VariablesMap
+import cacophony.semantic.syntaxtree.FunctionalExpression
+import kotlin.math.max
+
+abstract class CallableHandlerImpl(
+    private val analyzedFunction: AnalyzedFunction, // TODO: we need to change it so it also covers lambda expressions
```

Comment:
You can also describe the TODOs here so it's all in one place: https://github.com/mhorod/cacophony/issues/305